From 251ce41e7f514b1e2c04297bce8e10b72140a172 Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Mon, 8 Sep 2014 16:13:41 -0400
Subject: [ethernet] be2net: Call dev_kfree_skby_any instead of kfree_skb

Message-id: <1410192881-30702-19-git-send-email-ivecera@redhat.com>
Patchwork-id: 92432
O-Subject: [RHEL7.1 PATCH 18/78] benet: Call dev_kfree_skby_any instead of kfree_skb.
Bugzilla: 1088575
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>

BZ: #1088575

Upstream commit(s):
commit d8ec2c02caa3515f35d6c33eedf529394c419298
Author: Eric W. Biederman <ebiederm@xmission.com>
Date:   Tue Mar 11 14:19:50 2014 -0700

    benet: Call dev_kfree_skby_any instead of kfree_skb.

    Replace free_skb with dev_kfree_skb_any in be_tx_compl_process as
    which can be called in hard irq by netpoll, softirq context
    by normal napi polling, and in normal sleepable context
    by the network device close method.

    Signed-off-by: "Eric W. Biederman" <ebiederm@xmission.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/emulex/benet/be_main.c b/drivers/net/ethernet/emulex/benet/be_main.c
index 181ab5c..2bf6c65 100644
--- a/drivers/net/ethernet/emulex/benet/be_main.c
+++ b/drivers/net/ethernet/emulex/benet/be_main.c
@@ -1919,7 +1919,7 @@ static u16 be_tx_compl_process(struct be_adapter *adapter,
   queue_tail_inc(txq);
  } while (cur_index != last_index);
 
- kfree_skb(sent_skb);
+ dev_kfree_skb_any(sent_skb);
  return num_wrbs;
 }
 
-- 
1.7.1